src/include/interval_set.h

```diff
@@ -461,6 +515,24 @@ class interval_set {
     assert(&b != this);
     clear();
 
+    const interval_set *s, *l;
+
+    if (a.size() < b.size()) {
+      s = &a;
+      l = &b;
+    } else {
+      s = &b;
+      l = &a;
+    }
+
+    if (!s->size())
+      return;
+
+    if (l->size() / s->size() >= 10) {
```

Comment:
could you accompany this magic number with a comment explaining why we do it differently when the size ratio is greater or equal to 10?

Reply:
Yes, updated as follows:
```c++
    /*
     * Use the lower_bound algorithm for larger size ratios
     * where it performs better, but not for smaller size
     * ratios where sequential search performs better.
     */
    if (l->size() / s->size() >= 10) {
      intersection_size_asym(*s, *l);
      return;
    }
```